### PR TITLE
[4.0] Remove all references to old JPrefixed Class in /components/

### PR DESCRIPTION
**Pull request by PhilETaylor**
Code Review Removing Technical Debt.

Remove reference to old JPrefixed Class + other small cleanups of doc blocks. 

Nothing here is controversial and should be a quick test and merge.

These changes are manually made and reviewed, over several hours, and are restricted to the /components/ folder only.

This PR is not exhaustive.

### CODE REVIEW

**Comment by wilsonge** [comment]
Shall we go with this one for conflict resolving next?

**Comment by PhilETaylor** [comment]
Hahahaha I suck obviously

**Comment by PhilETaylor** [comment]
Done.

File: components/com_config/src/Controller/ModulesController.php [hunk 1]
```diff
@@ -31,14 +31,14 @@
 class ModulesController extends BaseController
 {
 	/**
-	 * Constructor.
+	 * @param   array                         $config   An optional associative array of configuration settings.
+	 *                                                  Recognized key values include 'name', 'default_task', 'model_path', and
+	 *                                                  'view_path' (this list is not meant to be comprehensive).
+	 * @param   MVCFactoryInterface|null      $factory  The factory.
+	 * @param   CMSApplication|null           $app      The Application for the dispatcher
+	 * @param   \Joomla\CMS\Input\Input|null  $input
```
Comment (wilsonge) [inline components/com_config/src/Controller/ModulesController.php]:
```suggestion
	 * @param   \Joomla\CMS\Input\Input|null  $input  The Input object for the request
```

File: components/com_config/src/Controller/TemplatesController.php [hunk 2]
```diff
@@ -27,14 +27,14 @@
 class TemplatesController extends BaseController
 {
 	/**
-	 * Constructor.
+	 * @param   array                         $config   An optional associative array of configuration settings.
+	 *                                                  Recognized key values include 'name', 'default_task', 'model_path', and
+	 *                                                  'view_path' (this list is not meant to be comprehensive).
+	 * @param   MVCFactoryInterface|null      $factory  The factory.
+	 * @param   CMSApplication|null           $app      The Application for the dispatcher
+	 * @param   \Joomla\CMS\Input\Input|null  $input
```
Comment (wilsonge) [inline components/com_config/src/Controller/TemplatesController.php]:
```suggestion
	 * @param   \Joomla\CMS\Input\Input|null  $input  The Input object for the request
```

File: components/com_contact/src/Controller/DisplayController.php [hunk 3]
```diff
@@ -23,20 +24,18 @@
...
-	 * 'view_path' (this list is not meant to be comprehensive).
-	 * @param   MVCFactoryInterface  $factory  The factory.
-	 * @param   CMSApplication       $app      The JApplication for the dispatcher
-	 * @param   \JInput              $input    Input
+	 * @param   array                         $config   An optional associative array of configuration settings.
+	 *                                                  Recognized key values include 'name', 'default_task', 'model_path', and
+	 *                                                  'view_path' (this list is not meant to be comprehensive).
+	 * @param   MVCFactoryInterface|null      $factory  The factory.
+	 * @param   CMSApplication|null           $app      The Application for the dispatcher
+	 * @param   \Joomla\CMS\Input\Input|null  $input
```
Comment (wilsonge) [inline components/com_contact/src/Controller/DisplayController.php]:
```suggestion
	 * @param   \Joomla\CMS\Input\Input|null  $input  The Input object for the request
```

File: components/com_content/src/Controller/DisplayController.php [hunk 4]
```diff
@@ -24,14 +25,12 @@
...
-	 * 'view_path' (this list is not meant to be comprehensive).
-	 * @param   MVCFactoryInterface  $factory  The factory.
-	 * @param   CMSApplication       $app      The JApplication for the dispatcher
-	 * @param   \JInput              $input    Input
+	 * @param   array                         $config   An optional associative array of configuration settings.
+	 *                                                  Recognized key values include 'name', 'default_task', 'model_path', and
+	 *                                                  'view_path' (this list is not meant to be comprehensive).
+	 * @param   MVCFactoryInterface|null      $factory  The factory.
+	 * @param   CMSApplication|null           $app      The Application for the dispatcher
+	 * @param   \Joomla\CMS\Input\Input|null  $input
```
Comment (wilsonge) [inline components/com_content/src/Controller/DisplayController.php]:
```suggestion
	 * @param   \Joomla\CMS\Input\Input|null  $input  The Input object for the request
```

File: components/com_fields/src/Controller/DisplayController.php [hunk 5]
```diff
@@ -25,18 +25,18 @@ class DisplayController extends \Joomla\CMS\MVC\Controller\BaseController
 	/**
 	 * Constructor.
 	 *
-	 * @param   array                $config   An optional associative array of configuration settings.
-	 * Recognized key values include 'name', 'default_task', 'model_path', and
-	 * 'view_path' (this list is not meant to be comprehensive).
-	 * @param   MVCFactoryInterface  $factory  The factory.
-	 * @param   CMSApplication       $app      The JApplication for the dispatcher
-	 * @param   \JInput              $input    Input
+	 * @param   array                		  $config   An optional associative array of configuration settings.
```
Comment (wilsonge) [inline components/com_fields/src/Controller/DisplayController.php]: This one's harder to fix by suggestions - but you seem to have a mix of tabs and spaces here according to the github diff

Comment (PhilETaylor) [reply components/com_fields/src/Controller/DisplayController.php]: serves me right for trying to edit on GitHub.com - I'll do this locally.

File: components/com_modules/src/Controller/DisplayController.php [hunk 6]
```diff
@@ -23,22 +25,20 @@
...
-	 * 'view_path' (this list is not meant to be comprehensive).
-	 * @param   MVCFactoryInterface  $factory  The factory.
-	 * @param   CMSApplication       $app      The JApplication for the dispatcher
-	 * @param   \JInput              $input    Input
+	 * @param   array                     $config   An optional associative array of configuration settings.
+	 * 										        Recognized key values include 'name', 'default_task', 'model_path', and
+	 * 										        'view_path' (this list is not meant to be comprehensive).
+	 * @param   MVCFactoryInterface|null  $factory  The factory.
+	 * @param   CMSApplication|null       $app      The Application for the dispatcher
+	 * @param   Input|null 	 			  $input
```
Comment (wilsonge) [inline components/com_modules/src/Controller/DisplayController.php]: Tabs again here

File: libraries/src/MVC/Controller/AdminController.php [hunk 7]
```diff
@@ -62,7 +62,7 @@ class AdminController extends BaseController
 	 *                                         Recognized key values include 'name', 'default_task', 'model_path', and
 	 *                                         'view_path' (this list is not meant to be comprehensive).
 	 * @param   MVCFactoryInterface  $factory  The factory.
-	 * @param   CMSApplication       $app      The JApplication for the dispatcher
+	 * @param   CMSApplication       $app      The Application for the dispatcher
 	 * @param   Input                $input    Input
```
Comment (wilsonge) [inline libraries/src/MVC/Controller/AdminController.php]:
```suggestion
	 * @param   Input                $input    Input  The Input object for the request
```

File: components/com_fields/src/Controller/DisplayController.php (hunk 5 above)
Comment (PhilETaylor) [reply components/com_fields/src/Controller/DisplayController.php]: ok?

Comment (wilsonge) [reply components/com_fields/src/Controller/DisplayController.php]: Yup looks better

**Comment by wilsonge** [comment]
Sorry one last rebase. I merged in the void return PR

**Comment by PhilETaylor** [comment]
ok I think Im finally done with this one 

1) Github lies... 
2) Github web interface sucks
3) Caching on Github web pages sucks
4) Nearly all the suggestions for Input you provided had tabs in them :) 
5) grrrrrr

**Comment by wilsonge** [comment]
Oops :( Sorry!

**Comment by PhilETaylor** [comment]
Now drone cs is not happy - will fix in morning if you don’t as I’m in bed now

**Comment by wilsonge** [comment]
Thanks!

**Comment by wilsonge** [comment]
I'll fix drone now - I trust myself more in IntelliJ than though the GitHub UI after this :D

**Pull request merged**